src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp

```diff
@@ -826,51 +904,22 @@ NetworkDescription LevelZeroCompilerInDriver<TableExtension>::compile(const std:
                     ". ",
                     getLatestBuildError());
 
-    // Get blob size first
-    size_t blobSize = -1;
-
-    result = _graphDdiTableExt->pfnGetNativeBinary(graphHandle, &blobSize, nullptr);
-
-    OPENVINO_ASSERT(result == ZE_RESULT_SUCCESS,
-                    "Failed to compile network. L0 pfnGetNativeBinary get blob size",
-                    " result: ",
-                    ze_result_to_string(result),
-                    ", code 0x",
-                    std::hex,
-                    uint64_t(result),
-                    ". ",
-                    getLatestBuildError());
-
-    std::vector<uint8_t> blob(blobSize);
-    // Get blob data
-    result = _graphDdiTableExt->pfnGetNativeBinary(graphHandle, &blobSize, blob.data());
-
-    OPENVINO_ASSERT(result == ZE_RESULT_SUCCESS,
-                    "Failed to compile network. L0 pfnGetNativeBinary get blob data",
-                    " result: ",
-                    ze_result_to_string(result),
-                    ", code 0x",
-                    std::hex,
-                    uint64_t(result),
-                    ". ",
-                    getLatestBuildError());
-
     auto networkMeta = getNetworkMeta(graphHandle);
     networkMeta.name = model->get_friendly_name();
 
-    result = _graphDdiTableExt->pfnDestroy(graphHandle);
+    _logger.debug("compile end");
 
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("Failed to compile network. L0 pfnDestroy",
-                       " result: ",
-                       ze_result_to_string(result),
-                       ", code 0x",
-                       std::hex,
-                       uint64_t(result));
-    }
+    auto networkDescription = NetworkDescription(std::vector<uint8_t>(), std::move(networkMeta));
 
-    _logger.debug("compile end");
-    return NetworkDescription(std::move(blob), std::move(networkMeta));
+    // Store the graph handle as a void pointer, return the same graphHandle to backend for ZeroExecutor
+    networkDescription.graphHandle = static_cast<void*>(graphHandle);
+
+    // Use const_cast to remove const qualifier
+    networkDescription.propsVoidPtr = const_cast<void*>(static_cast<const void*>(&_props));
+    networkDescription.inputDescriptors = const_cast<void*>(static_cast<const void*>(&inputDescriptors));
+    networkDescription.outputDescriptors = const_cast<void*>(static_cast<const void*>(&outputDescriptors));
+
+    return networkDescription;
 }
 
 template <typename TableExtension>
```

Comment:
We need the same changes for the import path, right?

Reply:
Also, the blob could be destroyed after the parse is done in the case of CiD only here:

https://github.com/openvinotoolkit/openvino/pull/26016/files#diff-f1f8f14a46f6bfbbfc9561e65e1a27b99d0e3a554964137190d414557eec728cR744

Reply:
Changed to share graphHandle also on import_model path now